src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -3303,10 +3303,22 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
             args.reserve(al, c->n_args);
             visit_expr_list(c->m_args, c->n_args, args);
             if (call_name == "print") {
-                ASR::expr_t *fmt=nullptr;
+                ASR::expr_t *fmt = nullptr;
                 Vec<ASR::expr_t*> args_expr = ASRUtils::call_arg2expr(al, args);
+                ASR::expr_t *separator = nullptr;
+                if (c->n_keywords > 0) {
+                    std::string arg_name;
+                    for (size_t i = 0; i < c->n_keywords; i++) {
+                        arg_name = c->m_keywords[i].m_arg;
+                        if (arg_name == "sep") {
+                            visit_expr(*c->m_keywords[i].m_value);
+                            separator = ASRUtils::EXPR(tmp);
+                            break;
```

Comment:
Let's check the type of `tmp`, make sure it's a string expression, otherwise give an error message, and let's write a test for the error message.